sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveShowCreateTableSuite.scala

```diff
@@ -61,129 +44,174 @@ class HiveShowCreateTableSuite extends ShowCreateTableSuite with TestHiveSinglet
+             |)
+             |STORED AS
+             |  INPUTFORMAT 'org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat'
+             |  OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat'
+           """.stripMargin
+        )
 
-  test("hive bucketing is supported") {
-    withTable("t1") {
-      sql(
-        s"""CREATE TABLE t1 (a INT, b STRING)
-           |CLUSTERED BY (a)
-           |SORTED BY (b)
-           |INTO 2 BUCKETS
-         """.stripMargin
-      )
-      checkCreateTable("t1")
+        checkCreateTable("t1")
+      }
     }
   }
 
-  test("hive partitioned view is not supported") {
-    withTable("t1") {
-      withView("v1") {
+  test("hive bucketing is supported") {
+    withSQLConf(
+        SQLConf.LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED.key -> "true") {
+      withTable("t1") {
         sql(
-          s"""
-             |CREATE TABLE t1 (c1 INT, c2 STRING)
-             |PARTITIONED BY (
-             |  p1 BIGINT COMMENT 'bla',
-             |  p2 STRING )
-           """.stripMargin)
-
-        createRawHiveTable(
-          s"""
-             |CREATE VIEW v1
-             |PARTITIONED ON (p1, p2)
-             |AS SELECT * from t1
+          s"""CREATE TABLE t1 (a INT, b STRING)
+             |CLUSTERED BY (a)
+             |SORTED BY (b)
+             |INTO 2 BUCKETS
            """.stripMargin
         )
+        checkCreateTable("t1")
+      }
+    }
+  }
 
-        val cause = intercept[AnalysisException] {
-          sql("SHOW CREATE TABLE v1")
+  test("hive partitioned view is not supported") {
+    withSQLConf(
+        SQLConf.LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED.key -> "true") {
+      withTable("t1") {
+        withView("v1") {
+          sql(
+            s"""
+               |CREATE TABLE t1 (c1 INT, c2 STRING)
+               |PARTITIONED BY (
+               |  p1 BIGINT COMMENT 'bla',
+               |  p2 STRING )
+             """.stripMargin)
+
+          createRawHiveTable(
+            s"""
+               |CREATE VIEW v1
+               |PARTITIONED ON (p1, p2)
+               |AS SELECT * from t1
+             """.stripMargin
+          )
+
+          val cause = intercept[AnalysisException] {
+            sql("SHOW CREATE TABLE v1")
+          }
+
+          assert(cause.getMessage.contains(" - partitioned view"))
         }
-
-        assert(cause.getMessage.contains(" - partitioned view"))
       }
     }
   }
 
   test("SPARK-24911: keep quotes for nested fields in hive") {
-    withTable("t1") {
-      val createTable = "CREATE TABLE `t1`(`a` STRUCT<`b`: STRING>) USING hive"
-      sql(createTable)
-      val shownDDL = getShowDDL("SHOW CREATE TABLE t1")
-      assert(shownDDL == createTable.dropRight(" USING hive".length))
+    withSQLConf(
+        SQLConf.LEGACY_CREATE_HIVE_TABLE_BY_DEFAULT_ENABLED.key -> "true") {
+      withTable("t1") {
+        val createTable = "CREATE TABLE `t1`(`a` STRUCT<`b`: STRING>) USING hive"
```

Comment:
Hm, doesn't this already creates a Hive table?

Reply:
Actually, I asked to cover all cases to avoid future parser changes at the other test cases (partitioned table), but @HyukjinKwon 's comment also makes sense at this case. @viirya . Sorry, you can omit these two cases.
- https://github.com/apache/spark/pull/27393#discussion_r372780566

Reply:
Oops, I missed that comment. What about we do it in beforeAll o beforeEach? I left a comment because it enables it for each test manually and it looked odd to do it for unnecessary cases as well.